cli/flox/src/commands/publish.rs

```diff
@@ -89,7 +89,7 @@ impl Publish {
             check_environment_metadata(&flox, &mut env).or_else(|e| bail!(e.to_string()))?;
 
         let build_metadata =
-            check_build_metadata(&env, &package, &flox.system).or_else(|e| bail!(e.to_string()))?;
+            check_build_metadata(&env, &package).or_else(|e| bail!(e.to_string()))?;
```

Comment:
suggestion: no need to explicitly bail, the result is already an error.

```suggestion
        let build_metadata = check_build_metadata(&env, &package)?;
```